tsl/src/compression/compression.c

```diff
@@ -2682,13 +2695,63 @@ process_predicates(Chunk *ch, CompressionSettings *settings, List *predicates, L
 																	 collation,
 																	 opcode,
 																	 arg_value,
-																	 false,	  /* is_null_check */
-																	 false)); /* is_null */
+																	 false, /* is_null_check */
+																	 false, /* is_null */
+																	 false	/* is_array_op */
+																	 ));
 						}
 					}
 				}
 			}
 			break;
+			case T_ScalarArrayOpExpr:
+			{
+				ScalarArrayOpExpr *sa_expr = castNode(ScalarArrayOpExpr, node);
+				if (!ts_extract_expr_args(&sa_expr->xpr, &var, &expr, &opno, &opcode))
+					continue;
+
+				if (!IsA(expr, Const))
```

Comment:
Might be simple to run in once on the entire predicate at the start of this loop, it's recursive.

Reply:
I'll leave this for another PR, it has the chance of resulting in more work that we can't benefit from, eg constraints on columns we cant filter on for compressed dml